src/fp/list.nim

```diff
@@ -114,6 +114,24 @@ proc foldRightF*[T, U](xs: List[T], z: () -> U, f: (T, () -> U) -> U): U =
   if xs.isEmpty: z()
   else: f(xs.head, () => xs.tail.foldRightF(z, f))
 
+proc unfoldLeft*[T](f: T -> Option[(T, T)], x:T): List[T] {. inline .}=
```

Comment:
There should be two type parameters here: http://hackage.haskell.org/package/base-4.9.1.0/docs/Data-List.html#v:unfoldr . The tests should also cover the case when the type arguments are different (e.g. unfolding a `string` to a list of `char`s).